### PR TITLE
Misc doc fixes

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
Misc doc fixes

### Types of change(s)
- [ ] Documentation update

### Checklist
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Documentation~/filter.yml [hunk 1]
```diff
@@ -8,6 +8,9 @@ apiRules:
 - exclude:
     uidRegex: ^Unity.MLAgents\.Tests\.Communicator$
     type: Namespace
+- exclude:
```
Comment (chriselion) [inline com.unity.ml-agents/Documentation~/filter.yml:11]: These were showing up in the API docs

File: com.unity.ml-agents/Runtime/Actuators/ActionSegment.cs [hunk 2]
```diff
@@ -107,19 +107,19 @@ public void Clear()
             System.Array.Clear(Array, Offset, Length);
         }
 
-        /// <inheritdoc cref="IEnumerable{T}.GetEnumerator"/>
+        /// <inheritdoc/>
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Actuators/ActionSegment.cs:110]: Got some warnings from the verbose doc generation about these.

File: com.unity.ml-agents/Runtime/Agent.cs [hunk 3]
```diff
@@ -1096,7 +1096,7 @@ public virtual void CollectObservations(VectorSensor sensor)
         /// <summary>
         /// Returns a read-only view of the observations that were generated in
         /// <see cref="CollectObservations(VectorSensor)"/>. This is mainly useful inside of a
-        /// <see cref="Heuristic(float[], int[])"/> method to avoid recomputing the observations.
+        /// <see cref="Heuristic(in ActionBuffers)"/> method to avoid recomputing the observations.
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Agent.cs:1099]: bad crefs

File: docs/Learning-Environment-Executable.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ over using the Editor:
 - You can exchange executable with other people without having to share your
   entire repository.
 - You can put your executable on a remote machine for faster training.
-- You can use `Headless` mode for faster training.
+- You can use `Server Build` mode for faster training.
```
Comment (vincentpierre) [inline docs/Learning-Environment-Executable.md:10]:
```suggestion
- You can use `Server Build` (`Headless`) mode for faster training (as long as the executable does not need rendering).
```

**Review by vincentpierre — APPROVED** [review]

**Pull request merged**